packages/runtime/src/modules/linker.ts

```diff
@@ -77,31 +91,38 @@ const linker: Linker = async (specifier, context, options = {}) => {
 
 // Loads a module as a general specifier or from a specific path
 const loadActualModule = async (specifier: string, options: LinkerOptions) => {
+  const log = options.log || defaultLogger;
+
   // Lookup the path from an explicit specifier first
-  let path = options.modulePaths?.[specifier] || null;
-  if (options.trace && path) {
-    console.log(`[linker] Loading module ${specifier} from mapped ${path}`);
+  let path;
+  let version;
+  if (options.modules?.[specifier]) {
+    ({ path, version } = options.modules?.[specifier]);
   }
 
   if (!path && options.repo) {
     // Try and load a matching path from the repo
-    path = await getModulePath(specifier, options.repo);
+    const specifierWithVersion = version
+      ? `${specifier}@${version}`
+      : specifier;
+    path = await getModulePath(specifierWithVersion, options.repo);
   }
 
   if (path) {
+    // TODO: should we log WHY this path was used? Maybe this is enough for now
+    log.debug(`[linker] Loading module ${specifier} from ${path}`);
     try {
       return import(path);
     } catch (e) {
       if (options.trace) {
-        console.warn(
-          `[linker] Failed to load module ${specifier} from ${path}`
-        );
+        log.debug(`[linker] Failed to load module ${specifier} from ${path}`);
```

Comment:
I'm curious about the shift from `warn` to `debug` (not the use of the logger), surely with these kinds of situation we should be logging above `debug`?

Reply:
It's because it's OK for this to "fail" and think a warning is too scary at this stage. if the module can't be found in the repo it'll do a regular import and pull in the local or global package, if found.

BUT I need to remove the outer `if (trace)`, because trace has been deprecated.  I managed to miss that somehow. And actually I should go ahead and remove trace properly.

There is another question here. Now that we prefer to load modules from the repo, I think we should probably disable the `import(specifier)` fallback. That will load globals or packages installed in the runtime, but otherwise fail. It's not very traceable if it goes wrong (or right).

But that's probably outside the scope of this, PR, so I'll spin out an issue

Reply:
Ok I've removed trace fully (thanks for the catch) and raised #78